### DISCUSSION TITLE
Maximum throughput

### DISCUSSION THREAD

**Discussion by improperaffo** [srsran/srsRAN_Project#256] [General Help]
Dear all,
we conducted some testing and we found out that our UEs do not achieve the full bit rate that should be possible for a given MCS and a given bandwidth. 

For example, using a TDD configuration over a 40 MHz band, allocating 4 slots to DL and 4 slots to UL, and splitting the mixed slots 6 symbol for DL and 6 for UL, leaving one slot and one symbol empty, (i.e., 44.6% resources allocated to DL and 44.6% to UL), using MCS table QAM256, and fixing MCS=14 (spectral efficiency 3.6094) for both DL and UL, we only achieve a bitrate of `41 Mbps` in DL and `29 Mbps` in UL. While we should be looking a something closer to `65 Mbps`. 
```
           -----------------DL-----------------------|------------------UL--------------------
 pci rnti  cqi  ri  mcs  brate   ok  nok  (%)  dl_bs | pusch  mcs  brate   ok  nok  (%)    bsr
   1 4602   15   1   14    41M 1000    0   0%   928k |  22.8   14   348k  128    0   0%      0
   1 4602   15   1   14    41M  998    2   0%  1.14M |  23.5   14   373k  110    0   0%     38
   1 4602   15   1   14    41M 1000    0   0%  1.22M |  23.0   14   342k  120    0   0%    384
   1 4602   15   1   14    41M  999    1   0%  1.34M |  21.9   14   250k  124    0   0%     28
   1 4602   15   1   14    41M 1000    0   0%  1.07M |  23.0   14   335k  125    0   0%      0
   1 4602   15   1   14   8.9M  220    0   0%      0 |  23.4   14   153k   42    0   0%      0
   1 4602   15   1   14    19k   10    0   0%      0 |  23.9   14   204k   11    0   0%      0
   1 4602   15   1   14   600k  112    0   0%    290 |  21.9   14   3.5M  102    0   0%   300k
   1 4602   15   1   14   759k  803    0   0%    174 |  21.1   14    29M  602    0   0%   413k
   1 4602   15   1   14   612k  752    0   0%    116 |  21.6   14    29M  600    0   0%   300k
   1 4602   15   1   14   581k  749    0   0%     58 |  21.7   14    29M  599    0   0%   300k
```

What could be the reason for these results? 

Moreover, why is the UL consistently lower than the DL, even allocating more resources to it, for same MCS, it is always performing poorer than the DL.

**Comment by ismagom** [comment]
Well, you are not taking into account the overhead in UL and DL. In the DL you have PDCCH, reference signals, SSB, etc. In the UL you have PUCCH, DMRS, etc.
